### PR TITLE
[HA] ensure server-provided fields are preserved in delta calculation

### PR DESCRIPTION
**Pull request by tom-vx51**
## What changes are proposed in this pull request?

This fixes a bug where server-provided data could be omitted on client-side labels, resulting in erroneous patch requests. Labels are now properly merged during delta calculations.

## How is this patch tested? If it is not, please explain why.

local, unit tests

Specific steps to repro:
* Create a new `Classification` field on a dataset
* Use the `edit_field_values` operator to init `<new_field>.label` to some value for all samples
* Edit the classification label in the annotate tab

Previous behavior:
* Persistence gets into a bad state due to missing local data

New behavior:
* Persistence works as expected

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

Fixed a bug which could cause erroneous patch requests while annotating samples

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

* **Bug Fixes**
  * Delta/patch generation for annotations updated to preserve existing server-enriched fields during partial updates.

* **Tests**
  * Expanded delta calculation test suite with broader edge-case coverage for nested, empty and mixed updates.

* **Chores**
  * Reorganized imports and utilities; added a shared object-check helper and cleaned up client-side imports.

### CODE REVIEW

File: app/packages/annotation/src/deltas.ts [hunk 1]
```diff
@@ -305,15 +305,28 @@ export const buildDeletionDeltas = (
...
+export const buildSingleMutationDelta = <
   T extends AnnotationLabel["data"] | Primitive
 >(
   sample: Sample,
   path: string,
   data: T
 ): JSONDeltas => {
   const existingLabel = <T>extractNestedField(sample, path) ?? {};
-  return generateJsonPatch(existingLabel, data);
+
```
Comment (erik-nieh) [inline app/packages/annotation/src/deltas.ts:316]: There is an existing isObject that might be useful, but I think it does not explicitly check the null case: 

https://github.com/voxel51/fiftyone/blob/develop/app/packages/core/src/client/util.ts#L89

**Review by erik-nieh — DISMISSED** [review]

File: app/packages/annotation/src/deltas.ts [hunk 2]
```diff
@@ -305,15 +305,28 @@ export const buildDeletionDeltas = (
...
+  //  (`_cls`, `_id`, `tags`) are preserved when the overlay only has a
+  //  subset of fields
+  const merged =
+    typeof data === "object" &&
+    data !== null &&
+    !Array.isArray(data) &&
+    typeof existingLabel === "object" &&
+    !Array.isArray(existingLabel)
+      ? { ...existingLabel, ...data }
+      : data;
```
Comment (mozabes) [inline app/packages/annotation/src/deltas.ts:323]: Is there no way for us to type this better than a generic T?  Can we be more explicit with the typing to make it so we can not have to be so verbose with checking typeofs and isArrays?

**Review by lanzhenw — DISMISSED** [review]
lgtm

File: app/packages/annotation/src/deltas.ts (hunk 2 above)
Comment (tom-vx51) [reply app/packages/annotation/src/deltas.ts:323]: yeah it's currently scoped to 
```typescript
T extends AnnotationLabel["data"] | Primitive
```
and the wrench here is `Primitive`. The guards are probably more than necessary, but it's important that we don't corrupt data here, so I'd rather have some extra caution. 

Per Erik's point, we can at least hide it in a utility function

File: app/packages/annotation/src/deltas.ts (hunk 1 above)
Comment (tom-vx51) [reply app/packages/annotation/src/deltas.ts:316]: updated, thanks!

**Review by lanzhenw — APPROVED** [review]
